docs/components/select.md

```diff
@@ -1,1013 +1,741 @@
+Label attributes to be passed to underlying [KLabel component](/components/label#props).
 
 <ClientOnly>
-  <KSelect label="Name" :label-attributes="{
-      help: 'I use the KLabel `help` prop',
-      'data-testid': 'test'
-    }"
-    :items="defaultItemsUnselect"
-  />
+  <KSelect :label-attributes="{ info: 'I use the KLabel `info` prop' }" label="Label" :items="selectItems" />
 </ClientOnly>
 
 ```html
-<KSelect
-  label="Name"
-  :label-attributes="{
-    help: 'I use the KLabel `help` prop',
-    'data-testid': 'test'
-  }"
-  :items="items"
-/>
+<KSelect :label-attributes="{ info: 'I use the KLabel `info` prop' }" label="Label" :items="selectItems" />
 ```
 
-### appearance
-
-There are three styles of selects, `select` and `dropdown` (default) which are filterable, and lastly `button` which is not.
+### placeholder
 
-The `dropdown` appearance style has a selected item object. You can deselect the item by clicking the Clear icon.
+Placeholder to be displayed when no item is selected.
 
 <ClientOnly>
-  <KSelect :items="deepClone(defaultItems)" />
+  <KSelect placeholder="Select service" :items="selectItemsUnselected" />
 </ClientOnly>
 
 ```html
-<KSelect :items="items" />
+<KSelect placeholder="Select service" :items="selectItems" />
 ```
 
-The `select` style displays the selected item in the text box and also displays a chevron. To allow deselecting the item, you need to
-set the `clearable` prop to `true`. See [clearable](#clearable) for an example.
-
-<ClientOnly>
-  <KSelect appearance="select" :items="deepClone(defaultItems)" />
-</ClientOnly>
-
-```html
-<KSelect appearance="select" :items="items" />
-```
+### clearable
 
-The `button` style triggers the dropdown on click and you cannot filter the entries.
+When set to `true`, the clear icon will be displayed in front of selected item.
 
 <ClientOnly>
-  <KSelect appearance="button" :items="deepClone(defaultItems)" />
+  <KSelect clearable :items="selectItems" />
 </ClientOnly>
 
 ```html
-<KSelect appearance="button" :items="items" />
+<KSelect clearable :items="selectItems" />
 ```
 
-### clearable
+### help
 
-The `clearable` prop is used to enable deselecting the selected item when `appearance` is `'select'`. Defaults to `false`.
+Help text to be displayed below the select element.
 
 <ClientOnly>
-  <KSelect appearance="select" :items="deepClone(defaultItems)" clearable />
+  <KSelect help="Helpful text." :items="selectItems" />
 </ClientOnly>
 
 ```html
-<KSelect appearance="select" :items="items" clearable />
+<KSelect help="Helpful text." :items="selectItems" />
 ```
 
-### buttonText
+### error
 
-You can configure the button text when an item is selected, if `appearance` is type `button`.
+When set to `true`, error styling will apply.
```

Comment:
The input element has an `errorMessage` prop, but this component reuses the `help` prop? Seems confusing - do we have a standard?

Reply:
Added a quick and easy way to sanitize user input: https://github.com/Kong/kongponents/pull/1832/commits/42cd593c12afc33cf797146d3074c346daf35c59. Though I would argue that this would need to be a separate ticket.